### PR TITLE
`Development`: Update Spring Boot to 3.4.0

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context
We want to use the latest versions to keep the code modern, stable and secure

### Description
Update Spring Boot and related dependencies to the latest version

### Steps for Testing
Go through the whole app and test all kinds of functionality. Everything should be exactly the same as before

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **New Features**
  - Enhanced mocking capabilities for Jenkins job permissions in various test classes.
  - New methods added to the `ProgrammingExerciseTestService` for importing programming exercises and managing repository access.

- **Bug Fixes**
  - Improved error handling in tests related to programming submissions and exercise creation.

- **Refactor**
  - Updated annotations from `@SpyBean` to `@MockitoSpyBean` across multiple classes to align with updated mocking practices.

- **Documentation**
  - Clarified guidelines and comments regarding the use of Mockito in tests.

### CODE REVIEW

**Review by SimonEntholzer — APPROVED** [review]
Tested exercises, exam mode participation and assessment, communication and clicked through several other components. Everything seems to be working as before

File: build.gradle [hunk 1]
```diff
@@ -212,10 +212,9 @@ repositories {
     maven {
         url "https://build.shibboleth.net/maven/releases"
     }
-    // required for latest jgit 7.0.0 dependency
-    // TODO: remove this when jgit is available in the official maven repository
+    // TODO: remove this when spring cloud is available in the official maven repository
```
Comment (AjayvirS) [inline build.gradle:215]: is this TODO still necessary?

**Comment by AjayvirS** [comment]
Tested exercise submissions, plagiarism detection, tutorials and learning path generation. The latter seems to trigger the following 500 error on initial request. Does not seem to be related to the upgrade though:
image

**Review by AjayvirS — APPROVED** [review]

**Review by HawKhiem — APPROVED** [review]
I tested on TS4 the followings: creating lectures, auto units generation, editing and appending units and attachments, communication functionalities like emojis, uploading images. creating competencies and editing them. Everything seems to work as before. I also tried adding new attachments and units to lectures and name them then try downloading them to see their names. The names of the downloaded attachments/units are not displayed as named. But this is covered in another PR #9899

**Pull request merged**